src/utils/common.ts

```diff
@@ -290,34 +293,65 @@ export async function estimateTotalGasRequiredByUnsignedTransaction(
 }
 
 /**
- * Create an unsigned transaction of a fillRelay contract call
- * @param spokePool The specific spokepool that will populate this tx
- * @param destinationTokenAddress A valid ERC20 token (system-wide default is UDSC)
- * @param simulatedRelayerAddress The relayer address that relays this transaction
- * @returns A populated (but unsigned) transaction that can be signed/sent or used for estimating gas costs
+ * Create an unsigned transaction to fill a relay. This function is used to simulate the gas cost of filling a relay.
+ * @param spokePool A valid SpokePool contract instance
+ * @param fillToSimulate The fill that this function will use to populate the unsigned transaction
+ * @returns An unsigned transaction that can be used to simulate the gas cost of filling a relay
  */
-export async function createUnsignedFillRelayTransaction(
+export async function createUnsignedFillRelayTransactionFromDeposit(
   spokePool: SpokePool,
-  destinationTokenAddress: string,
-  simulatedRelayerAddress: string
+  deposit: Deposit,
+  amountToFill: BN,
+  relayerAddress: string,
+  _relayerBalanceForToken?: BN
 ): Promise<PopulatedTransaction> {
-  // Populate and return an unsigned tx as per the given spoke pool
-  // NOTE: 0xBb23Cd0210F878Ea4CcA50e9dC307fb0Ed65Cf6B is a dummy address
-  return await spokePool.populateTransaction.fillRelay(
-    "0xBb23Cd0210F878Ea4CcA50e9dC307fb0Ed65Cf6B",
-    "0xBb23Cd0210F878Ea4CcA50e9dC307fb0Ed65Cf6B",
-    destinationTokenAddress,
-    "10",
-    "10",
-    "1",
-    "1",
-    "1",
-    "1",
-    "1",
-    [],
-    MAX_BIG_INT,
-    { from: simulatedRelayerAddress }
-  );
+  // We should check that the relayer has enough balance to facilitate this
+  // transaction before we populate it.
+  const relayerBalanceForToken =
+    _relayerBalanceForToken ?? (await getTokenBalance(relayerAddress, deposit.originToken, spokePool.provider));
+  if (relayerBalanceForToken.lt(amountToFill)) {
+    throw new Error(
+      `Relayer balance for token (${relayerBalanceForToken.toString()}) is less than the amount to fill (${amountToFill.toString()})`
+    );
+  }
+
+  // If we have made it this far, then we can populate the transaction.
+  if (isDefined(deposit.speedUpSignature)) {
+    return spokePool.populateTransaction.fillRelayWithUpdatedDeposit(
+      deposit.depositor,
+      deposit.recipient,
+      deposit.updatedRecipient ?? "0x",
+      deposit.destinationToken,
+      amountToFill,
```

Comment:
this needs to be swapped with below param. In order, the params are `amount` and then `maxTokensToSend` which are the full deposit amount and then the actual amount to relay